src/Statement/StatementList.php

```diff
@@ -282,4 +282,24 @@ public function isEmpty() {
 		return empty( $this->statements );
 	}
 
+	/**
+	 * @since 3.0
+	 *
+	 * @param Statement $statement
+	 *
+	 * @return int|bool
+	 */
+	public function indexOf( Statement $statement ) {
+		$index = 0;
+
+		foreach ( $this->statements as $s ) {
+			if ( $s->equals( $statement ) ) {
```

Comment:
`removeStatementWithGuid` and `hasStatementWithGuid` for by GUID and `remove` and `contains` for value based comparison seems to make more sense to me. `indexOf` falling into the later category. Perhaps we need the by GUID index lookup, this has still not been demonstrated to me though. Yes, if you change the value then you won't find it. Do we have a use case where this is a problem?